azure-pipelines.yml

```diff
@@ -45,13 +52,19 @@ steps:
   - script: |
       set -Eeuo pipefail
       docker login -u="$(QUAY_USERNAME)" -p="$(QUAY_PASSWORD)" quay.io
-      
       docker tag keboola/docker-custom-python:python-3.8 quay.io/keboola/$(imageRepository_base):python-3.8-$(imageTag)
+      docker push -a quay.io/keboola/$(imageRepository_base)
+    displayName: Push 3.8 image to quay.io
+    condition: and(eq(variables['isTagBuild'], 'true'), eq(variables['changedProjects_python38'], 1))
+
+  - script: |
+      set -Eeuo pipefail
+      docker login -u="$(QUAY_USERNAME)" -p="$(QUAY_PASSWORD)" quay.io
       docker tag keboola/docker-custom-python:python-3.10 quay.io/keboola/$(imageRepository_base):python-3.10-$(imageTag)
       docker tag keboola/docker-custom-python:python-3.10 quay.io/keboola/$(imageRepository_base):latest
       docker push -a quay.io/keboola/$(imageRepository_base)
```

Comment:
If I'm not mistaken, all `script` runs on a single host/environment, so this will try push again also the 3.8 image already pushed above. I know I'm complicating stuff again :slightly_smiling_face: but maybe I would trurn this back to a single `script` for all versions and just use `if` to tag nessecary versions based on variables and then push them at once with `docker push -a`.

Or, throw away `docker push -a` and push tags one by one.

Reply:
This of course applies also to ACR